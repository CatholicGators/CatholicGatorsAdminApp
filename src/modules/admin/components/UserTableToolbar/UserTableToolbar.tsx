import React from "react"
import {
  createStyles,
  withStyles,
  Toolbar,
  Typography,
  Tooltip,
  IconButton
} from "@material-ui/core";
import DoneIcon from '@material-ui/icons/Done'
import DeleteIcon from '@material-ui/icons/Delete';
import VpnKey from '@material-ui/icons/VpnKey';

const styles = createStyles({
    toolbarTitle: {
        marginRight: 'auto'
    },  
})

type Props = {
    classes: any,
    numSelected: number,
    handleBatchApprove: () => any,
    handleBatchAuthorize: () => any,
    handleBatchDelete: () => any
}

export const UserTableToolbar = (props: Props) => {
    const {
        classes,
        numSelected,
        handleBatchApprove,
        handleBatchAuthorize,
        handleBatchDelete
    } = props

    return (
        <Toolbar>
            <div className={classes ? classes.toolbarTitle : null}>
                {numSelected > 0 ? (
                    <Typography color="inherit" variant="subtitle1">
                        {numSelected} selected
                    </Typography>
                    ) : (
                    <Typography variant="h6">
                        Users
                    </Typography>
                )}
            </div>
            {numSelected > 0 ? (
                <div>
                    <Tooltip title="Approve">
                        <IconButton
                            onClick={() => handleBatchApprove()}
                            aria-label="Approve"
                        >
                            <DoneIcon />
                        </IconButton>
                    </Tooltip>
                    <Tooltip title="Make admin">
                        <IconButton
                            onClick={() => handleBatchAuthorize()}
                            aria-label="Make admin"
                        >
                            <VpnKey />
                        </IconButton>
                    </Tooltip>
                    <Tooltip title="Delete">
                        <IconButton
                            aria-label="Delete"
                            onClick={() => handleBatchDelete()}
                        >
                            <DeleteIcon />
                        </IconButton>
                    </Tooltip>
                </div>
            ) : null}
        </Toolbar>
    );
}

export default withStyles(styles)(UserTableToolbar)
