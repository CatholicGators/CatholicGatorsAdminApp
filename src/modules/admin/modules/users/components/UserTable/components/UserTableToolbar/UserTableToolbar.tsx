import React from "react";
import {
    createStyles,
    withStyles,
    Toolbar,
    Typography,
    Tooltip,
    IconButton,
    Theme,
} from "@material-ui/core";
import DoneIcon from "@material-ui/icons/Done";
import DeleteIcon from "@material-ui/icons/Delete";
import VpnKey from "@material-ui/icons/VpnKey";

export const styles = (theme: Theme) =>
    createStyles({
        toolbarTitle: {
            marginRight: "auto",
        },
    });

type Props = {
    classes: any;
    numSelected: number;
    handleBatchApprove: () => any;
    handleBatchAuthorize: () => any;
    handleBatchDelete: () => any;
};

export const UserTableToolbar = (props: Props) => {
    const {
        classes,
        numSelected,
        handleBatchApprove,
        handleBatchAuthorize,
        handleBatchDelete,
    } = props;

    return (
        <Toolbar>
            <div className={classes ? classes.toolbarTitle : null}>
                {numSelected > 0 ? (
                    <Typography
                        id="num-selected"
                        color="inherit"
                        variant="subtitle1"
                    >
                        {numSelected} selected
                    </Typography>
                ) : (
                    <Typography id="title" variant="h6">
                        Users
                    </Typography>
                )}
            </div>
            {numSelected > 0 ? (
                <div>
                    <Tooltip title="Approve">
                        <IconButton
                            id="approve-icon-btn"
                            onClick={() => handleBatchApprove()}
                            aria-label="Approve"
                        >
                            <DoneIcon />
                        </IconButton>
                    </Tooltip>
                    <Tooltip title="Make admin">
                        <IconButton
                            id="authorize-icon-btn"
                            onClick={() => handleBatchAuthorize()}
                            aria-label="Make admin"
                        >
                            <VpnKey />
                        </IconButton>
                    </Tooltip>
                    <Tooltip title="Delete">
                        <IconButton
                            id="delete-icon-btn"
                            aria-label="Delete"
                            onClick={() => handleBatchDelete()}
                        >
                            <DeleteIcon />
                        </IconButton>
                    </Tooltip>
                </div>
            ) : null}
        </Toolbar>
    );
};

export default withStyles(styles)(UserTableToolbar);
