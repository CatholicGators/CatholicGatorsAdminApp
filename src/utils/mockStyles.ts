import { createMuiTheme } from '@material-ui/core'

export default (styles) => Object.keys(styles(createMuiTheme())).reduce((acc, curr) => ({...acc, [curr]: "test"}), {})