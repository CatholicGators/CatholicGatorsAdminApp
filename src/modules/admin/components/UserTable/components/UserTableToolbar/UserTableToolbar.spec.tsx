import React from 'react'
import { shallow } from 'enzyme';
import { UserTableToolbar, styles } from './UserTableToolbar';
import { IconButton } from '@material-ui/core';
import mockStyles from '../../../../../../utils/mockStyles';

describe('UserTableToolbar', () => {
    let wrapper, props;

    beforeEach(() => {
        props = {
            classes: mockStyles(styles),
            numSelected: 0,
            handleBatchApprove: jest.fn(),
            handleBatchAuthorize: jest.fn(),
            handleBatchDelete: jest.fn()
        }
        wrapper = shallow(<UserTableToolbar {...props}/>)
    })

    it('matches snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    })

    it('hides action buttons when nothing selected', () => {
        expect(wrapper.exists(IconButton)).toBe(false)
    })

    it('shows the title when nothing selected', () => {
        expect(wrapper.find('#title').exists()).toBe(true)
        expect(wrapper.find('#num-selected').exists()).toBe(false)
    })

    it('shows the action buttons when something is selected', () => {
        wrapper.setProps({ numSelected: 1 })

        expect(wrapper.exists(IconButton)).toBe(true)
    })

    it('shows the numSelected when > 0', () => {
        wrapper.setProps({ numSelected: 1 })

        expect(wrapper.find('#title').exists()).toBe(false)
        expect(wrapper.find('#num-selected').exists()).toBe(true)
    })

    it('calls handleBatchApprove() when the approve IconButton is clicked', () => {
        wrapper.setProps({ numSelected: 1 })
        wrapper.find('#approve-icon-btn').simulate('click')

        expect(props.handleBatchApprove).toHaveBeenCalled()
    })

    it('calls handleBatchAuthorize() when the authorize IconButton is clicked', () => {
        wrapper.setProps({ numSelected: 1 })
        wrapper.find('#authorize-icon-btn').simulate('click')

        expect(props.handleBatchAuthorize).toHaveBeenCalled()
    })

    it('calls handleBatchDelete() when the delete IconButton is clicked', () => {
        wrapper.setProps({ numSelected: 1 })
        wrapper.find('#delete-icon-btn').simulate('click')

        expect(props.handleBatchDelete).toHaveBeenCalledWith()
    })
})