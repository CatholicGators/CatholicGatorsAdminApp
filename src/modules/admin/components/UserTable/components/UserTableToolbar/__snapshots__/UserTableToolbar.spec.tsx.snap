// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserTableToolbar matches snapshot 1`] = `
<WithStyles(ForwardRef(Toolbar))>
  <div
    className="test"
  >
    <WithStyles(ForwardRef(Typography))
      id="title"
      variant="h6"
    >
      Users
    </WithStyles(ForwardRef(Typography))>
  </div>
</WithStyles(ForwardRef(Toolbar))>
`;
